lib/backend/k8s/resources/customresource.go

```diff
@@ -319,7 +345,12 @@ func (c *customK8sResourceClient) Watch(ctx context.Context, list model.ListInte
 	toKVPair := func(r Resource) (*model.KVPair, error) {
 		return c.convertResourceToKVPair(r)
 	}
-	return newK8sWatcherConverter(ctx, resl.Kind+" (custom)", toKVPair, k8sWatch), nil
+
+	v1v3converter := func(r Resource) (Resource, error) {
```

Comment:
Urm, looking at the common threads in this code...

Looks like putting the conversion in convertResourceToKVPair will handle all of your use cases.

Reply:
I saw that first but it feels like we're sneaking in this thing that's not obvious from reading the method name `convertResourceToKVPair` which I didn't like

Reply:
also, `convertResourceToKVPair` is also used in Create/Update, don't we want to limit this conversion to Get/List/Watch?